### PR TITLE
Bug 1961091: Adding GatherMachineHealthCheck

### PR DESCRIPTION
**Pull request by rluders**
This PR implements a new data enhancement to gather `machinehealthcheck`.

## Categories

- [ ] Bugfix
- [X] Enhancement
- [ ] Backporting
- [ ] Others (CI, Infrastructure, Documentation)

## Sample Archive

- `docs/insights-archive-sample/config/machinehealthchecks/openshift-machine-api/machine-api-termination-handler.json`

## Documentation

- `docs/gathered-data.md`

## Unit Tests

- `pkg/gatherers/clusterconfig/machine_healthcheck_test.go`

## Privacy

Yes. There are no sensitive data in the newly collected information.

## Breaking Changes

No

## References

https://issues.redhat.com/browse/CCXDEV-4633
https://bugzilla.redhat.com/show_bug.cgi?id=1961091

### CODE REVIEW

File: docs/gathered-data.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 This document is auto-generated by `make gen-doc`
 
+## 3Records
```
Comment (0sewa0) [inline docs/gathered-data.md:3]: wut is this ?

File: docs/gathered-data.md [hunk 2]
```diff
@@ -437,6 +490,14 @@ Docs for relevant types: https://pkg.go.dev/github.com/openshift/api/operatorcon
   * 4.8+
 
 
+## Panic
+
+
+
+## Period
```
Comment (0sewa0) [inline docs/gathered-data.md:497]: ??

File: docs/gathered-data.md [hunk 3]
```diff
@@ -437,6 +490,14 @@ Docs for relevant types: https://pkg.go.dev/github.com/openshift/api/operatorcon
   * 4.8+
 
 
+## Panic
```
Comment (0sewa0) [inline docs/gathered-data.md:493]: ????

File: docs/gathered-data.md [hunk 4]
```diff
@@ -237,6 +253,10 @@ Response see https://docs.okd.io/latest/rest_api/machine_apis/containerruntimeco
   * 4.7+
 
 
+## Errors
```
Comment (0sewa0) [inline docs/gathered-data.md:256]: ???

File: docs/gathered-data.md [hunk 5]
```diff
@@ -316,6 +357,10 @@ Gathered metrics:
 * Id in config: metrics
 
 
+## Name
```
Comment (0sewa0) [inline docs/gathered-data.md:360]: ???

File: docs/gathered-data.md [hunk 6]
```diff
@@ -535,6 +598,14 @@ Response see https://docs.openshift.com/container-platform/4.3/rest_api/index.ht
   * 4.7+
 
 
+## ShouldBeProcessed
+
+
+
+## SomeField
```
Comment (0sewa0) [inline docs/gathered-data.md:605]: ???

**Comment by 0sewa0** [comment]
other then the weird stuff in the `gathered-data.md`, it looks good

**Comment by rluders** [comment]
@0sewa0 done, removed all the missing entries.

**Pull request merged**